spec/controllers/queue_items_controller_spec.rb

```diff
@@ -81,32 +77,117 @@
     end
   end
 
-  describe "DELETE destory" do 
-    it "should redirect to my queue page" do
-      session[:user_id] = Fabricate(:user).id
-      queue_item = Fabricate(:queue_item)
-      delete :destroy, id: queue_item.id
-      expect(response).to redirect_to my_queue_path
+  describe "DELETE destory" do
+    context "with authenticated user" do
+
+      let(:thupten) { Fabricate(:user)}
+
+      before { session[:user_id] = thupten.id }
+
+      it "should redirect to my queue page" do
+        queue_item = Fabricate(:queue_item)
+        delete :destroy, id: queue_item.id
+        expect(response).to redirect_to my_queue_path
      end
+
      it "delete the queue item" do 
-        thupten = Fabricate(:user)
-        session[:user_id] = thupten.id 
         queue_item = Fabricate(:queue_item, user: thupten)
         delete :destroy, id: queue_item.id
         expect(QueueItem.count).to eq(0)
      end
+
      it "does not delete the queue item if the current user does not own" do
-        thupten = Fabricate(:user)
         kevin = Fabricate(:user)
-        session[:user_id] = thupten.id
         kevin_queue_item = Fabricate(:queue_item, user: kevin)
         delete :destroy, id: kevin_queue_item.id
         expect(QueueItem.count).to eq(1) 
      end
-     it "redirects to sign in page for unauthenticated user" do 
-       delete :destroy, id: 3
-       expect(response).to redirect_to sign_in_path 
+
+     it "normalizes the remaining queue_items after destroy" do 
+        queue_item1 = Fabricate(:queue_item, user: thupten, position: 1)
+        queue_item2 = Fabricate(:queue_item, user: thupten, position: 2)
+        delete :destroy, id: queue_item1.id
+        expect(QueueItem.first.position).to eq(1)
      end
+    end
+
+    context "with unauthenticated user" do
+      it "redirects to sign in page for unauthenticated user" do 
+      delete :destroy, id: 3
+      expect(response).to redirect_to sign_in_path 
+     end
+    end     
+  end
+
+  describe "POST update_queue" do
+    context "with valid input" do
+
+      let(:queue_item1) { Fabricate(:queue_item, user: don, video: video, position: 1) }
+      let(:queue_item2) { Fabricate(:queue_item, user: don, video: video, position: 2) }
+      let (:don) { Fabricate(:user) }
+      let(:video) { Fabricate(:video)}
+
+      before { session[:user_id] = don.id }
+
+      it "should redirect to my queue page" do
```

Comment:
I would recommend to avoid using word "should" in examples message - it does not add any value, rather use 3rd person singular for verb - Read more [here](http://www.betterspecs.org/#should)